internal/gateway/proxy/proxy.go

```diff
@@ -87,6 +138,260 @@ type ControllerProxy struct {
 	grpcSrv      *grpc.Server
 	ctrl         cluster.Cluster
 	writerMap    sync.Map
+	cache        sync.Map
+}
+
+func (cp *ControllerProxy) Publish(ctx context.Context, req *vanuspb.PublishRequest) (*emptypb.Empty, error) {
```

Comment:
can merge this with https://github.com/linkall-labs/vanus/blob/main/internal/gateway/proxy/proxy.go#L92 ?

Reply:
no problem